lib/monitoring/aws-docdb/DocumentDbMetricFactory.ts

```diff
@@ -0,0 +1,121 @@
+import { DimensionsMap } from "aws-cdk-lib/aws-cloudwatch";
+import { DatabaseCluster } from "aws-cdk-lib/aws-docdb";
+
+import {
+  getLatencyTypeLabel,
+  getLatencyTypeStatistic,
+  LatencyType,
+  MetricFactory,
+  MetricStatistic,
+} from "../../common";
+
+const DocumentDbNamespace = "AWS/DocDB";
+
+export interface DocumentDbMetricFactoryProps {
+  /**
+   * database cluster identifier (either this or `cluster` need to be specified)
+   */
+  readonly clusterIdentifier?: string;
+  /**
+   * database cluster (either this or `clusterIdentifier` need to be specified)
+   */
+  readonly cluster?: DatabaseCluster;
+}
+
+export class DocumentDbMetricFactory {
+  readonly clusterIdentifier: string;
+  protected readonly metricFactory: MetricFactory;
+  protected readonly dimensionsMap: DimensionsMap;
+
+  constructor(
+    metricFactory: MetricFactory,
+    props: DocumentDbMetricFactoryProps
+  ) {
+    this.metricFactory = metricFactory;
+    this.clusterIdentifier =
+      DocumentDbMetricFactory.resolveDbClusterIdentifier(props);
+    this.dimensionsMap = { DBClusterIdentifier: this.clusterIdentifier };
+  }
+
+  private static resolveDbClusterIdentifier(
+    props: DocumentDbMetricFactoryProps
+  ): string {
+    if (props.clusterIdentifier !== undefined && props.cluster === undefined) {
+      return props.clusterIdentifier;
+    } else if (
+      props.cluster !== undefined &&
+      props.clusterIdentifier === undefined
+    ) {
+      return props.cluster.clusterIdentifier;
+    } else {
+      throw new Error(
+        "Specify either 'clusterIdentifier' or 'cluster', but not both."
+      );
```

Comment:
This else block also covers the case where neither of them are specified. The message in that case would be misleading.
We should add two different error messages when both or neither are provided

Reply:
OK - I thought it was clear enough, but we can raise the bar here and be very specific.

Reply:
cool, will do